File: ruby/p5.rb
#!/usr/bin/ruby

def gcd(x,y)
	if y == 0 
		return x
	else
		return gcd(y,x%y)
	end

end

def lcm(x,y)
	return (x*y)/gcd(x,y)
end

x=1

for i in 1..20
	x=lcm(x,i)
	print "#{i}, #{x}\n"
end

File: python/p5.py
#find the least common multiple of all numbers 1-20

	
def gcd(a,b):
	if b == 0:
		return a
	else:
		return gcd(b, a % b)


def lcm(a,b):
	return (a*b)/gcd(a,b)

x = 1

for i in range(1,21):
	x = lcm(x,i)
	print i

print x

